docs/Learning/dependencies.md

```diff
@@ -0,0 +1,83 @@
+# Dependencies
+
+This documentation provides more detailed information about SELF's dependencies.
+
+
+## Message Passing Interface (MPI)
+
+### Background
+In order to support scaling of conservation law solvers to large supercomputers, SELF exposes additional parallelism through domain decomposition. This strategy divides a physical domain into multiple subdomains and gives ownership of each subdomain to a single process. These processes are often called "MPI ranks" or "MPI processes". So long as the SELF program you write enables domain decomposition during mesh creation, launching with `mpirun -np 4 /path/to/your/program` will provide you with domain decomposition across four MPI ranks. 
+
+With domain decomposition, each process has its own private memory. However, your conservation law likely demands that neighboring elements in your mesh participate in flux calculations together. When two neighboring elements reside on two separate processes, information must be shared. In SELF, this is achieved as part of the `SideExchange` methods for each data type when domain decomposition is enabled.
+
+On CPU-only platforms, the `boundary` attributes of the `SELF_Mapped*` types are passed to `MPI_ISEND` (asynchronous send) to send data to neighboring processes. On the flip size, the `extboundary` attributes are passed to the `MPI_IRECV` (asynchronous recv) to receive data from the neighboring processes. With each `MPI_ISEND/IRECV` pair, a unique tag for the message is calculated using the global edge/face ID and variable ID. This ensures the messages get stored in the correct address locations when they are received.
```

Comment:
Is asterisk intentional in `SELF_Mapped*`? Do you mean "flip side" instead of "flip size"?

Reply:
Asterisk is intentional.. we have a variety of "Mapped" data types... e.g. "MappedScalar_1d, MappedScalar_2d, etc...

Yes, flip side and not flip size.. oops.

Reply:
Fixed in [3738412](https://github.com/FluidNumerics/SELF/pull/53/commits/37384125cd5dd1e4542576d1b53b8a362304e2a2)